### PR TITLE
Utilize multifile compiling

### PR DESCRIPTION
**Pull request by blitz-1306**
## Dependencies
- This PR depends on #162 (and should be reviewed/merged after it).

## Changes
- [x] Tweaked Scribble executable to utilize multifile compiling from **solc-typed-ast**.
- [x] Removed source unit merging logic - this is done by compiler on its own.
- [x] Updated affected test samples.

## Notes
- **Multifile compiling is supported only for `source` input mode.** The `json` input mode assumes that all necessary sources are already in the input file. Merging multiple JSON files does not seem to be a great idea due to possible key collistions and amount of possibe input scenarios. So there is still requirement to have a single input file when input mode is set to `json`.
- I suggest to pay additional attention to changed flattened samples. Some node ids was changed. Seems to be "okay" but I'm not completely sure that it really is "okay".

Regards.

### CODE REVIEW

File: src/bin/scribble.ts [hunk 1]
```diff
@@ -151,8 +150,17 @@ function ppWarning(warn: Warning): string[] {
     return [`${ppLoc(start)} Warning: ${warn.msg}`, getSrcLine(warn.location)];
 }
 
+function assertIsFile(fileName: string): void {
+    assert(fileName !== "", "Path is empty");
+    assert(fse.existsSync(fileName), "Path not found: " + fileName);
+
+    const stats = fse.statSync(fileName);
```
Comment (cd1m0) [inline src/bin/scribble.ts:157]: This is not a blocker - just a small optimization. We can remove the call to `fse.existsSync` here by doing:
```
assert(fileName !== "", "Path is empty");
try {
        const stats = fse.statSync(fileName);
        assert(stats.isFile(), "Target is not a file: " + fileName);
} catch (e) {
         assert(false, e.code === 'ENOENT' ? ("Path not found: " + fileName) :  e.message);
}
```

File: src/bin/scribble.ts [hunk 2]
```diff
@@ -891,44 +887,21 @@ function pickVersion(versionUsedMap: Map<string, string>): string {
...
-                isSane(mergedUnit, mergedCtx),
-                `Merged unit ${mergedUnit.absolutePath} is insane`
-            );
-        }
-
-        for (const target of targets) {
-            const units = groupsMap.get(target) as SourceUnit[];
-            const files = filesMap.get(target) as Map<string, string>;
+        for (const unit of units) {
+            assert(isSane(unit, astCtx), `Merged unit ${unit.absolutePath} is insane`);
```
Comment (cd1m0) [inline src/bin/scribble.ts:892]: I don't think we need this assert anymore

**Review by cd1m0 — CHANGES_REQUESTED** [review]
Good job! Love commits that delete code. Two small nits - one is not a blocker

File: src/bin/scribble.ts (hunk 1 above)
Comment (blitz-1306) [reply src/bin/scribble.ts]: Addressed in https://github.com/ConsenSys/scribble/pull/164/commits/42b350e912bd4eef2f59dc60d3bdfcb6f27e67d3

File: src/bin/scribble.ts (hunk 2 above)
Comment (blitz-1306) [reply src/bin/scribble.ts]: Addressed in https://github.com/ConsenSys/scribble/pull/164/commits/42b350e912bd4eef2f59dc60d3bdfcb6f27e67d3

**Review by cd1m0 — APPROVED** [review]
LGTM

**Pull request merged**